### PR TITLE
[Music][Cleanup]Fix some old compiler warnings in MusicDatabase

### PR DESCRIPTION
**Pull request by DaveTBlake**
Fix some compiler warnings that were reported here
https://jenkins.kodi.tv/view/Reports/job/LINUX-64-GL-Static-Analysis/lastSuccessfulBuild/cppcheck/fileName.-489090852/New/#blamesContent

Mostly 
- Passing the result of c_str() to a function that takes std::string
- Reducing scope of variables
- Removeal of unused variables

### CODE REVIEW

File: xbmc/music/MusicDatabase.cpp [hunk 1]
```diff
@@ -11919,7 +11912,7 @@ void CMusicDatabase::ExportToXML(const CLibExportSettings& settings,
               }
             }
             xmlDoc.Clear();
-            TiXmlDeclaration decl("1.0", "UTF-8", "yes");
+            // TiXmlDeclaration decl("1.0", "UTF-8", "yes");
```
Comment (lrusak) [inline xbmc/music/MusicDatabase.cpp:11915]: What's up with this?

File: xbmc/music/MusicDatabase.cpp [hunk 2]
```diff
@@ -12062,7 +12054,7 @@ void CMusicDatabase::ExportToXML(const CLibExportSettings& settings,
                 }
               }
               xmlDoc.Clear();
-              TiXmlDeclaration decl("1.0", "UTF-8", "yes");
+              // TiXmlDeclaration decl("1.0", "UTF-8", "yes");
```
Comment (lrusak) [inline xbmc/music/MusicDatabase.cpp:12057]: same here

File: xbmc/music/MusicDatabase.cpp [hunk 3]
```diff
@@ -7600,7 +7594,7 @@ static const translateJSONField JSONtoDBSong[] = {
   { "displayconductor",          "string", false, "Role_Conductor",         "song_artist.idRole AS Role_Conductor" },
   { "displayorchestra",          "string", false, "Role_Orchestra",         "song_artist.idRole AS Role_Orchestra" },
   { "displaylyricist",           "string", false, "Role_Lyricist",          "song_artist.idRole AS Role_Lyricist" },
-
+
```
Comment (lrusak) [inline xbmc/music/MusicDatabase.cpp:7597]: extra space

File: xbmc/music/MusicDatabase.cpp [hunk 4]
```diff
@@ -7458,14 +7452,14 @@ bool CMusicDatabase::GetAlbumsByWhereJSON(
                   StringUtils::Split(record->at(1 + i).get_asString(), ",");
               if (values.size() % 2 == 0) // Must contain an even number of entries
               {
-                for (size_t i = 0; i + 1 < values.size(); i += 2)
+                for (size_t j = 0; j + 1 < values.size(); j += 2)
                 {
-                  int idGenre = atoi(values[i].c_str());
+                  int idGenre = atoi(values[j].c_str());
```
Comment (lrusak) [inline xbmc/music/MusicDatabase.cpp:7457]: not really applicable but this could be `std::stoi` then you can drop the `c_str()`

**Review by lrusak — COMMENTED** [review]
Seems sane :upside_down_face:

File: xbmc/music/MusicDatabase.cpp (hunk 1 above)
Comment (DaveTBlake) [reply xbmc/music/MusicDatabase.cpp:11915]: It is a repeated definition, initialised so far up in the code I wanted to have a comment as a reminder of how it is set. I guess better to add a comment that looks less like code

File: xbmc/music/MusicDatabase.cpp (hunk 2 above)
Comment (DaveTBlake) [reply xbmc/music/MusicDatabase.cpp:12057]: Same reason, a reminder of what type of thing `decl` is and how it is initialised

File: xbmc/music/MusicDatabase.cpp (hunk 3 above)
Comment (DaveTBlake) [reply xbmc/music/MusicDatabase.cpp:7597]: I swear they secretly breed in my dev env while I'm not looking

File: xbmc/music/MusicDatabase.cpp (hunk 4 above)
Comment (DaveTBlake) [reply xbmc/music/MusicDatabase.cpp:7457]: Want the silent fail of `atoi` not the exception raised by `stoi`

**Review by lrusak — APPROVED** [review]
LGTM, thanks for doing the cleanup :)

**Pull request merged**